src/Dev/Tasks/TagsToShortcodeHelper.php

```diff
@@ -0,0 +1,290 @@
+                        $this->rewriteFieldForRecords($records, $table, $field);
+                    } catch (DatabaseException $exception) {
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Takes a set of query results and updates image urls within a page's content.
+     * @param Query $records
+     * @param string $updateTable
+     * @param string $field
+     */
+    private function rewriteFieldForRecords(Query $records, string $updateTable, string $field)
+    {
+        foreach ($records as $row) {
+            $content = $row[$field];
+            $newContent = $this->getNewContent($content);
+            if ($content == $newContent) {
+                continue;
+            }
+
+            $updateSQL = SQLUpdate::create($updateTable)->addWhere(['"ID"' => $row['ID']]);
+            $updateSQL->addAssignments(["\"$field\"" => $newContent]);
+            $updateSQL->execute();
+            DB::alteration_message('Updated page with ID ' . $row['ID'], 'changed');
+        }
+    }
+
+    /**
+     * @param string $content
+     * @return string
+     */
+    private function getNewContent(string $content)
+    {
+        $tags = $this->getTagsInContent($content);
+        foreach ($tags as $tag) {
+            if ($newTag = $this->getNewTag($tag)) {
+                $content = str_replace($tag, $newTag, $content);
+            }
+        }
+
+        return $content;
+    }
+
+    /**
+     * Get all tags within some page content and return as array.
+     * @param $content
+     * @return array
+     */
+    private function getTagsInContent($content)
+    {
+        $resultTags = [];
+
+        preg_match_all('/<('.$this->validTagsPattern.').*?('.$this->validAttributesPattern.')\s*=.*?>/', $content, $matches, PREG_SET_ORDER);
+        if ($matches) {
+            foreach ($matches as $match) {
+                $resultTags []= $match[0];
+            }
+        }
+
+        return $resultTags;
+    }
+
+    /**
+     * @param string $tag
+     * @return array
+     */
+    private function getTagTuple(string $tag)
+    {
+        $pattern = sprintf(
+            '/.*<(?<tagType>%s).*(?<attribute>%s)="(?<src>[^"]*)"/i',
+            $this->validTagsPattern,
+            $this->validAttributesPattern
+        );
+        preg_match($pattern, $tag, $matches);
+        return $matches;
+    }
+
+    /**
+     * Gets the value of a given attribute from a tag.
+     * @param $tag
+     * @param $attribute
+     * @return null
+     */
+    private static function getAttribute($tag, $attribute)
+    {
+        $propertyValue = null;
+        $needle = $attribute . '="';
+        if (strpos($tag, $needle)) {
+            $propertyValue = explode('"', explode($needle, $tag)[1])[0];
+        }
+
+        return $propertyValue;
+    }
+
+    /**
+     * Extracts an array of attributes from a tag.
+     * @param string $tag
+     * @param string $tagType
+     * @param $id
+     * @param $newSrc
```

Comment:
```suggestion
     * @param string $newSrc
```

`$id` is not on the method definition